source/slang/syntax.cpp

```diff
@@ -1841,6 +1866,16 @@ void Type::accept(IValVisitor* visitor, void* extra)
         return nullptr;
     }
 
+    RefPtr<Substitutions> substituteSubstitutions(RefPtr<Substitutions> oldSubst, Substitutions * subst, int * ioDiff)
+    {
+        if (oldSubst)
+            oldSubst = oldSubst->SubstituteImpl(subst, ioDiff);
+
+        RefPtr<Substitutions> newSubst = oldSubst;
+        insertGlobalGenericSubstitutions(newSubst, subst, ioDiff);
```

Comment:
Ah, I think I see. The call to `insertGlobalGenericSubstititions()` will modify things in-place, so it isn’t safe to call on a substitution object in the middle of the list.

Eventually we can fix this up so that the insertion of the global generic substa is just the NULL case here, and I think it will Just Work. Not a priority for now.